File: ruby/testcode/benchmark_test_00619.rb
require_relative 'shared'
require 'logger'

def log_integer_safe(req)
  user_id = req.param('user_id').to_i
  logger = Logger.new($stdout)
  logger.info("Login user_id=#{user_id}")
  BenchmarkResponse.ok('ok')
end

File: rust/testcode/benchmark_test_00619.rs
pub fn handle(req: &super::shared::BenchmarkRequest) -> super::shared::BenchmarkResponse {
    let rows: usize = req.param("rows").parse().unwrap_or(0);
    let cols: usize = req.param("cols").parse().unwrap_or(0);
    let size = rows.checked_mul(cols);
    match size {
        Some(s) if s <= 1_000_000 => super::shared::BenchmarkResponse::ok(&format!("size={}", s)),
        Some(_) => super::shared::BenchmarkResponse::bad_request("Too large"),
        None => super::shared::BenchmarkResponse::bad_request("Overflow"),
    }
}
